grpc-circuitbreaker-utils/src/main/java/org/hypertrace/circuitbreaker/grpcutils/CircuitBreakerConfiguration.java

```diff
@@ -0,0 +1,16 @@
+package org.hypertrace.circuitbreaker.grpcutils;
+
+import java.util.Map;
+import java.util.function.BiFunction;
+import lombok.Builder;
+import lombok.Value;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+
+@Value
+@Builder
+public class CircuitBreakerConfiguration<T> {
+  Class<T> requestClass;
+  BiFunction<RequestContext, T, String> keyFunction;
+  boolean enabled;
+  Map<String, CircuitBreakerThresholds> circuitBreakerThresholdsMap;
```

Comment:
We should probably have a `CircuitBreakerThresholds` - the map being just for any custom overrides (which would be supported but atypical).